system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -4922,16 +4934,16 @@ class CLdapClient : implements ILdapClient, public CInterface
         LdapServerType stype = m_ldapconfig->getServerType();
         if(stype == ACTIVE_DIRECTORY)
         {
-            groupdn.append("cn=Administrators,cn=Builtin,").append(m_ldapconfig->getBasedn());
-        }
-        else if(stype == IPLANET)
-        {
-            groupdn.append("cn=Directory Administrators,").append(m_ldapconfig->getBasedn());
+            if (0 == stricmp(m_ldapconfig->getAdminGroupName(), "Administrators"))
+                groupdn.append("cn=Administrators,cn=Builtin,").append(m_ldapconfig->getBasedn());
+            else
+                groupdn.appendf("cn=%s,%s", m_ldapconfig->getAdminGroupName(), m_ldapconfig->getGroupBasedn());
```

Comment:
getGroupBasedn() or getBasedn()?

Reply:
Administrators is an LDAP system group and is stored in a different location (baseDN) than user groups (GroupsBaseDN)